packages/x-tree-view/src/hooks/useTreeViewApiRef.tsx

```diff
@@ -1,10 +1,10 @@
 import * as React from 'react';
-import { TreeViewAnyPluginSignature, TreeViewUsedPublicAPI } from '../internals/models';
+import { TreeViewAnyPluginSignature, TreeViewPublicAPI } from '../internals/models';
+import { DefaultTreeViewPlugins } from '../internals';
 
 /**
  * Hook that instantiates a [[TreeViewApiRef]].
  */
 export const useTreeViewApiRef = <
-  T extends TreeViewAnyPluginSignature,
-  Api extends TreeViewUsedPublicAPI<T>,
->() => React.useRef(undefined) as React.MutableRefObject<Api>;
+  TPlugins extends readonly TreeViewAnyPluginSignature[] = DefaultTreeViewPlugins,
```

Comment:
The typing was weird, it was only taking a single plugin
Now by default the type is the one of the default plugins (which are used by both the rich and the simple tree view)
but you can pass a generic to support more plugins (not useful for now, will become useful with RichTreeView-only plugins if they end up having public methods)